synapse/lib/syntax.py

```diff
@@ -67,11 +70,68 @@ def parse_literal(text, off, trim=True):
     return parse_int(text, off, trim=trim)
 
 def parse_int(text, off, trim=True):
-    numstr, off = nom(text, off, intset, trim=trim)
-    try:
-        return int(numstr, 0), off
-    except Exception as e:
-        raise s_common.BadSyntaxError(expected='Literal', at=off, got=text[off:off + 10])
+
+    _, off = nom(text, off, whites)
+
+    neg = False
+    if nextchar(text, off, '-'):
+        neg = True
+        _, off = nom(text, off + 1, whites)
+
+    valu = None
+    if nextstr(text, off, '0x'):
+        valu, off = nom(text, off + 2, hexset)
+        if not valu:
+            raise s_common.BadSyntaxError(at=off, mesg='0x expected hex')
+        valu = int(valu, 16)
+
+    elif nextstr(text, off, '0b'):
+        valu, off = nom(text, off + 2, binset)
+        if not valu:
+            raise s_common.BadSyntaxError(at=off, mesg='0b expected bits')
+        valu = int(valu, 2)
+
+    else:
+
+        valu, off = nom(text, off, decset)
+        if not valu:
+            raise s_common.BadSyntaxError(at=off, mesg='expected digits')
+
+        if not nextchar(text, off, '.'):
+            valu = int(valu)
+
+        else:
+            frac = nom(text, off + 1, decset)
+            valu = float('%s.%s' % (valu, frac))
```

Comment:
Should `parse_int` be parsing floats?

Reply:
detecting floats ( from the token parser perspective ) is kinda a subset of int since you basically parse an int and see if it has . and more digits.  May be worth separating this out eventually.